### DISCUSSION TITLE
My (gruelling) 3.5K item Homebox inventory - project details for those curious

### DISCUSSION THREAD

**Discussion by danielrosehill** [sysadminsmedia/homebox#163] [Show and tell]
Hi everyone,

Firstly, I'm really glad to see that the project has some new traction behind it. 

I shared a few Reddit posts and blogs about my (crazy) home inventory project. 

As much as it at times sucked (not the software, the process), it's proven immensely valuable. I've added about 3.5K items to my Homebox inventory which I realise makes my inventory a kind of "can this scale?" proof of concept.

In response to that all I can say is that I have (very happily) not noticed any deterioration in performance as the numbers have increased. And of course compared to what even a medium sized company might inventorise it's peanuts. But yes ... very nice system. 

## Using NFC Tags

![image](https://github.com/user-attachments/assets/aae00844-3ee3-44b9-b771-c419315e2715)

![Inventory20240803_041906](https://github.com/user-attachments/assets/2f252429-d201-4569-9ae2-87dbaee89fcc)

## Using QR Coes (Actually, Data Matrix)

![image](https://github.com/user-attachments/assets/92d27acf-320d-4456-84eb-2e5abe0174b2)

![image](https://github.com/user-attachments/assets/a129865f-a771-4c3f-97c9-cbed2891aefd)

![image](https://github.com/user-attachments/assets/081b3312-2567-41db-aa16-3264e4987054)

![image](https://github.com/user-attachments/assets/45e3212d-0f0e-415d-ad59-65203b4991de)

![image](https://github.com/user-attachments/assets/0109fc57-ae8a-4502-a97e-5ebc98515fee)

![image](https://github.com/user-attachments/assets/b088fe7d-9ceb-4dea-b120-8a83481fda76)

![image](https://github.com/user-attachments/assets/62ab02c8-e366-435e-b1cf-427cf4048b4b)

![image](https://github.com/user-attachments/assets/fe205a0d-f8af-44f0-8f0f-b4aa8aa8cf15)

## Process Outline

Initially I installed Homebox on a Synology NAS.

Later I realised that Synology NASes are very constraining in terms of hardware upgrades (failed RAM upgrade). So I converted an old desktop as a server and installed Homebox on it. 

To print the labels I picked up a Brother QL700. The QL800 or whatever comes next are much better options. 

Then I bought a bunch of continuous length tape in a few sizes (DK2210 and DK2205 or similar). 

I used SyncThingy to quickly photograph stuff and sync it over to my desktop. At my desktop I entered the inventory. 

For printing the labels .... I used a template in Ptouch Editor (the Brother label printing software). I think it's important to print the barcode AND the inventory number. If the barcode fades and becomes illegible (quite easy with sun exposure) you have a backup.

Later, I got into NFC and began writing the asset URLs onto NFC tags. This is now what I mostly do. The downside is that NFC tags may not hold data for "decades" but hey ... who knows what will happen then. The upside is that there's a wide variety of NFC tags on the market and you can get them on places that it's hard to affix labels to.

## Networking Info

To access the inventory remotely I set up a Cloudflare Tunnel and static routing (I think) so that the local IP structure could work outside of the LAN.

The easier option would have been to write the URLs on a real TLD (e.g. dansinventory.com). But I didn't want to predicate my whole inventory on maintaining a domain name (forget to renew the domain and all your stuff can no longer be found etc).

## How To Handle Boxes

Initially I configured all my boxes as 'locations' but then quickly realised that was a bad idea:

- You can't add photos to locations
-  You can't easily call up an inventory of the storage boxes

For this reason I spent a very unpleasant day manually scraping off stickers of box "locations" and recreating them as assets. One feature that would be useful in this respect would be the ability to convert asset types in a way that preserved the URL.

## Hardware I Bought For This

- Brother QL700
- Lots of label tape
- An NFC reader/writer 
- NFC tags
- Caffeine pills (LOL). But actually. This was a gruelling endeavor. 

## Benefits, Tweaks

While the process was tedious it has been extremely helpful.

As I went through my inventory I added a few labels:

- To give away
- To try out
- Duplicate purchases

I was able to quickly see instances in which I had accidentally "overpurchased" and bucket up those items in a giveaway list to donate to charity. 

I was also able to identify things I had purchased but never gotten around to actually using and flag those. 

My new "workflow" -- in general, I kind of naturally reached a point at which I "had everything I needed" for my tech consulting business, but naturally items periodically need to be replaced or new things purchased.

Whenever I get something (e.g. I just picked up a new USB headset) I will:

- Photograph it
- Make sure to photo any serial numbers / PNs etc
- Commit all that data into Homebox

The only reason I'm still printing labels is because I'm a sorry adherent to the "sunk cost" fallacy. But overall I've come to find NFC tags much more versatile and helpful. And I bought a few different colors so that I can try to make them as discrete as possible (my wife is less keen on the "we live in a warehouse" vibe but has proven a surprise supporter of the system).

I recently visited the US for a few weeks, and it was *super* useful to be able to remotely check my home inventory to see if I needed something from Amazon. This kind of got me excited again in the system and made me realise that while it was a huge amount of work in the end it really helped.

About me: love tech and have a huge and growing collection of tech stuff. Also live in a modestly sized apartment. Also have adult ADHD and tend to be very creative but highly disorganised. Homebox has proven an absolute gem in terms of helping me get control of the "stuff" I own. 

(Am learning programming ATM and a "down the line" aspiration is to actually become a technical contributor to the project!)

## Some Screenshots

![image](https://github.com/user-attachments/assets/39ee7917-65f1-4bb4-8e5d-77f699f7e4da)

![image](https://github.com/user-attachments/assets/c3df32b3-a06f-43e5-aea2-812bc062f7ed)

![image](https://github.com/user-attachments/assets/d5dec14b-8deb-4973-a69f-a47c42bce588)

![image](https://github.com/user-attachments/assets/80f532a5-1c35-4360-a515-004601da9fcc)

![image](https://github.com/user-attachments/assets/35bbb730-c837-4d09-98ce-a27e9b9e6865)

**Comment by bbobrian** [comment]
How did you create sub-items within Box 020 in your sample screenshot?

**Reply by danielrosehill** [reply]
@bbobrian The UI is a little clunky in this respect. IIRC I had to create the box as a parent item and then nest stuff in it. 

Looking back I have absolutely no idea how I did this. But weirdly enough it seems to have released me from buying crap that I don't need (I guess I saw how quickly "cheap" tech becomes obsolete and began wondering how much wastage I was responsible for).

I've been much more deliberate in my purchasing since and whenever I pick up something new that I care about, I oddly look forward to taking a few pics and putting it into the inventory. 

(Sorry, you didn't ask, but the notification made me think about the project for the first time in a while!)

**Reply by bbobrian** [reply]
Yea I figured it out. You can't do it when you create an item/asset, but when you edit there is a toggle for Advanced that will then let you set a Parent Item.

**Comment by tankerkiller125** [comment]
Damn is this impressive, but I still think we can do better ;) gatta get those images optimized and we 100% need to sort out the label generator and make that better.

**Reply by danielrosehill** [reply]
LOL. I used a separate Android app because I had so many pics to take (I kept them low res before I moved this over to a machine with a lot of storage headroom).

**Reply by Saturo78** [reply]
Same here. I have nearly 5.000 items at the moment and still growing.

**Reply by Saturo78** [reply]
which app do you use to take the photos and create the csv file?

**Reply by crispybegs** [reply]
> which app do you use to take the photos and create the csv file?

would also like to know this

**Reply by danielrosehill** [reply]
> > which app do you use to take the photos and create the csv file?
> 
> would also like to know this

 I used Open Camera on Android

**Reply by crispybegs** [reply]
that's a very useful app, thank you

**Comment by danielrosehill** [comment]
Yup! It's not the most intuitive but it gets the job done. Still better
than setting them up as locations!

Daniel Rosehill
danielrosehill.com

(Written from my phone, please excuse typos!)

**Reply by niyisurvey** [reply]
Hey there, Daniel. I don't think the world knows what a wonderful solution this is, and what an impact it potentially could have on people's lives. If we all manage to  get this app and workflow perfect, the number of people it could help would be amazong. I am thinking of people with brain injuries etc (myself). 

If you knew what my house looked like.. i couldn't even see the floor. I used to spend literally hours trying to find things and for about two years, everyone just thought I was a hoarder? I am not. The issue was that I just couldn't find anything with my 'new' brain. And also, when looking for something for more than 15 minutes, my brain fatigue kicked in, and I was just like a zombie forgetting I had looked in a box just 15 seconds before already.

The house looked awful, stopped dating, wouldn't let my friends in the house... then I wouldn't let my family in the house even !

Using this app, I have finally started to get my life back. I started properly at the end of last year and I hopefully will be finished in the next week or so. can you believe I am doing a PhD at Cambridge as well? Which is why I opened with talking about the impact of the app and your suggested workflow. Helping someone with the 'simple' things (well, simple to everyone else. even walking rehabilitation I guess could fall under this category) empowers them to do the big stuff which /actually/ matter, if you get me?

But anyway, I have a question. I have done QR codes to the URL of the location.. why should the nfc or qr code be an asset? Why should the location be an asset? I don't get why that's better.

Thanks again for your hard work.. and you SHOULD be reminded regularly how amazing this system is and what you took your time to methodically do and document for the rest of us !! :)

**Reply by niyisurvey** [reply]
Hold on a second, you say "You can't easily call up an inventory of the storage boxes" — what does that mean? Do you mean you want to know how many boxes you have all together? Is that the benefit?

Or looking closely at your pictures...you search for an item in homebox, and a label tells you the box (asset) associated with it?

And if you scan a box what happens then?

**Comment by danielrosehill** [comment]
Please excuse the very late response!

This is one of the most heartwarming messages I've ever received on GitHub!

I'm usually in a tech frame of mind whenever I post on GitHub. But for me
too homebox has been a totally transformational experience.

I'm transparent about being a highly creative but entirety scatterbrained
ADHD patient.

It's taken me into my mid thirties to appreciate that whatever systems make
my life better are nothing to be ashamed of.

Actually, I think I've finally reached the point where I don't really care
what people think.

Pretty much everything I own has an NFC tag and gets logged in HB. It's
still kind of a pain in the backside. But it's a weirdly meditative
process. It makes me appreciate what I have. And best of all I get to
actually make use of the things that I purchased.

Thank you so much for your delightful message. It brings me so much
happiness to know this others share the same sense of gratitude and warmth
towards the project.

We're all just internet strangers. But I like to think that Homeboxers must
be a good bunch!


Daniel Rosehill
danielrosehill.com

Note: This message was written from my phone, possibly using speech to text
software. Please ignore or reinterpret any obvious typos missed!

**Comment by mygrexit** [comment]
Hello @danielrosehill 

I just stopped by briefly because I read your message, and it really helped me set up my system exactly like yours. It also saved me from making a mistake by adding the storage bins as locations instead of items.

But somehow, your posts also touched me because I also have adult ADHD and struggle with the exact same issues as you. I’m extremely creative but also extremely unorganized, scattered, and very forgetful. I often forget what I own, buy things again even though I already have them, and sometimes end up with four or five of the same item lying around. I really enjoy going through my things and knowing what I have and what I don’t.

I don’t know why, but I felt very understood and seen by your post. It’s comforting to know that I’m not alone in this. I just wanted to thank you for that and wish you all the best.

**Comment by agryson** [comment]
Thanks for the walkthrough - I have a question though, before I spend several weekends and evenings cataloguing my workshop - could you expand on the pros/cons of using assets instead of locations for boxes/bins?

What do you mean by "You can't easily call up an inventory of the storage boxes"?
When I go to a location I see all items in that location down the bottom or is this not what you mean?

**Reply by agryson** [reply]
~~Also, you say you used data matrices, but I can only see QR codes in HB, I assume you did this manually rather than me missing a configuration option somewhere?~~
Nevermind, I see elsewhere that you did this manually.

**Reply by agryson** [reply]
Unless I'm missing something I think I might stick with locations for physical containers or areas and keep assets for items or consumables. 

For example, I have a 3D printer physically located in the office, but the consumables and spare parts are stored in different places, like the workshop.

So by sticking to physical place=location and thing=item, I can list the spare parts as being "in" box A but a "child item" of the printer so if I need a part or consumable I can find it quickly from the HB page of the printer. 

In my tests using assets as boxes I couldn't set up that kind of relation unless I had a box specifically for that thing... Which I probably could do, but seems off?

Happy for input if I'm misunderstanding anything!

**Reply by danielrosehill** [reply]
> ~Also, you say you used data matrices, but I can only see QR codes in HB, I assume you did this manually rather than me missing a configuration option somewhere?~ Nevermind, I see elsewhere that you did this manually.

Sorry for the slow reply - I was out hiking most of yday and away from screens etc. 

I think the main reason I did it like this was that locations don't have photos (still) and I have a large collection of boxes that sometimes look the same.

I think the most logical implementation would be to have a separate category all together for storage units (so: locations, items, boxes). But until/if that's implemented ... I think the parent and child items is a decent workaround.

So my systemin full is basically:

- Rooms (locations)  ... which have
- Boxes (parent items) .... which have  
- Items 

As well as loose items.

## Other Things

I use locations also as general units to nest items.

For example I have a 'location' for my car so that I can note what stuff I keep there.

And a parent unit called 'Offsite Locations' and then subunits for 'In Laws' House' etc.

**Reply by agryson** [reply]
Thanks for the reply, and hope you enjoyed the hike!
I will definitely use the car/somewhere else location idea, but given my use case of consumables/parts etc. (which seems a little different to yours) I might stick with naming boxes/containers as locations since I don't have the images as a requirement.
Have to say, thanks for all you content on this, it's really helped me get started!

**Reply by danielrosehill** [reply]
@agryson  Surely!

My inventory is mostly: computer parts, camera gear, more computer parts ... and sprinkles of random crap that I added to the inventory (see: food jars) because I was already committed to this crazy project and I went a bit overboard.

These days it's in maintenance mode and I just add a few things a week as I pick them up and lately get rid of some stuff, write the giveaway/disposal date as a "task" and then archive it. Working nicely.
